TheMET/TheMET/Network/MetAPI.swift

```diff
@@ -19,28 +19,29 @@ class MetAPI{
         self.networkManager = networkManager
     }
     
-    func objects(metadataDate: Date? = nil, departmentIds: [Int] = [], completion: @escaping (ObjectsResponse?) -> Void) {
-        self.metAPICache.objects(metadataDate: metadataDate, departmentIds: departmentIds) { [weak self] objectsResponce in
+    func objects(metadataDate: Date? = nil, departmentIds: [Int] = [], completion: @escaping (Result<ObjectsResponse, MetAPIError>) -> Void) {
+        self.metAPICache.objects(metadataDate: metadataDate, departmentIds: departmentIds) { [weak self] objectsResponse in
             guard let self = self else {
-                completion(nil)
+                completion(.failure(.metAPIDoesNotExist))
                 return
             }
-            if let objectsResponce = objectsResponce {
-                completion(objectsResponce)
+            if let objectsResponse = objectsResponse {
+                completion(.success(objectsResponse))
             } else {
-                self.executeObjects(metadataDate: metadataDate, departmentIds: departmentIds) {[weak self] realResponse in
-                    guard let realResponse = realResponse else {
-                        completion(nil)
-                        return
+                self.executeObjects(metadataDate: metadataDate, departmentIds: departmentIds) {[weak self] realResponseResult in
+                    switch realResponseResult {
+                    case .failure:
+                        completion(.failure(.noDataInResponse))
```

Comment:
This is not a `noDataInResponse`, use the error associated with the `failure` case:
```
case .failure(let error):
```